lib/searchkick/results.rb

```diff
@@ -221,6 +221,37 @@ def misspellings?
       @options[:misspellings]
     end
 
+    def scroll_id
+      @response['_scroll_id']
+    end
+
+    def scroll
+      if scroll_id.present? && options[:scroll].nil?
+        raise Searchkick::Error, "Scroll error - scroll keepalive must be defined"
+      elsif scroll_id.nil?
+        raise Searchkick::Error, "Scroll error - a scroll id has not been provided"
+      else
+        begin
+          params = {
+            scroll: options[:scroll],
+            scroll_id: scroll_id
+          }
+          es_results = Searchkick.client.scroll(params)
+          if scroll_id != es_results['_scroll_id']
+            # update the scroll id so we can keep reusing the same result set
+            @response['_scroll_id'] = es_results['_scroll_id'] 
```

Comment:
I see how this makes the API cleaner, but don't think we should mutate the original response right now. Can revisit at a later time if needed.